### PR TITLE
[gamenetworkingsockets] update to 1.3.0

### LINKED ISSUE #18312: [gamenetworkingsockets] update to 1.3.0
Library name: gamenetworkingsockets

New version number: 1.3.0

https://github.com/ValveSoftware/GameNetworkingSockets/releases/tag/v1.3.0

### PR DESCRIPTION
**Pull request by aizuon**
- #### What does your PR fix?  
  Fixes #18312

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Same with previous version

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]
Please add "vcpkg_fixup_pkgconfig" below "vcpkg_fixup_cmake_targets" in portfile.cmake

File: ports/gamenetworkingsockets/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "gamenetworkingsockets",
-  "version-date": "2021-04-26",
+  "version-string": "1.3.0",
```
[the remark this answers is not in the document]
Comment (aizuon) [reply ports/gamenetworkingsockets/vcpkg.json:3]: Thanks, I wasn't sure which string to use

File: versions/g-/gamenetworkingsockets.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "ba37a5a3ebed307b49a8174147bb3b142e3d18a6",
+      "version-string": "1.3.0",
```
[the remark this answers is not in the document]
Comment (aizuon) [reply versions/g-/gamenetworkingsockets.json:5]: Does vcpkg use the string declared in ports' vcpkg.json here when x-add-version is ran?

**Comment by aizuon** [comment]
Is the usage of vcpkg_fixup_pkgconfig correct without any arguments? I've checked the buildtrees of gamenetworkingsockets and couldn't find and .pc files.

File: versions/g-/gamenetworkingsockets.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (JonLiu1993) [reply versions/g-/gamenetworkingsockets.json]: > Does vcpkg use the string declared in ports' vcpkg.json here when x-add-version is ran?

If we change the "version string" to "version", after running "x-add-version", _versions/g-/gamenetworkingsockets.json_ will be automatically changed to "version"

**Comment by JonLiu1993** [comment]
> Is the usage of vcpkg_fixup_pkgconfig correct without any arguments? I've checked the buildtrees of gamenetworkingsockets and couldn't find and .pc files.

[This ](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_fixup_pkgconfig.md)is the official reference document of vcpkg_fixup_pkgconfig. You can refer to it. I found the .pc file in the buildtrees of gamenetworkingsockets before. I have re-run CI, and I will check it again after running.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
> > Is the usage of vcpkg_fixup_pkgconfig correct without any arguments? I've checked the buildtrees of gamenetworkingsockets and couldn't find and .pc files.
> 
> [This ](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_fixup_pkgconfig.md)is the official reference document of vcpkg_fixup_pkgconfig. You can refer to it. I found the .pc file in the buildtrees of gamenetworkingsockets before. I have re-run CI, and I will check it again after running.

After rerun, I also tested locally and did not find the .pc file, the files below it are all correct

**Comment by strega-nil-ms** [comment]
Cool, thanks @aizuon!

**Pull request merged**